pkgs/by-name/ma/material-black-colors/package.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, colorVariants ? []
+}:
+
+let
+  pname = "material-black-colors";
+
+in
+lib.checkListOfEnum "${pname}: color variants" [ "MB-Blueberry-Suru-GLOW" "MB-Cherry-Suru-GLOW" "MB-Lime-Suru-GLOW" "MB-Mango-Suru-GLOW" "MB-Pistachio-Suru-GLOW" "MB-Plum-Suru-GLOW" "Material-Black-Blueberry-Numix-FLAT" "Material-Black-Blueberry-Numix" "Material-Black-Blueberry-Suru" "Material-Black-Cherry-Numix-FLAT" "Material-Black-Cherry-Numix" "Material-Black-Cherry-Suru" "Material-Black-Lime-Numix-FLAT" "Material-Black-Lime-Numix" "Material-Black-Lime-Suru" "Material-Black-Mango-Numix-FLAT" "Material-Black-Mango-Numix" "Material-Black-Mango-Suru" "Material-Black-Pistachio-Numix-FLAT" "Material-Black-Pistachio-Numix" "Material-Black-Pistachio-Suru" "Material-Black-Plum-Numix-FLAT" "Material-Black-Plum-Numix" "Material-Black-Plum-Suru" ] colorVariants
+
+stdenvNoCC.mkDerivation rec {
+  inherit pname;
+  version = "unstable-2020-12-17";
+
+  src = fetchFromGitHub {
+    owner = "rtlewis88";
+    repo = "rtl88-Themes";
+    rev = "3864d851aac7f4e76cf23717aee104de234aef74";
+    hash = "sha256-BUJMd6Ltq16/HqqDbB5VDGIRSzLivXxNYZPT9sd6oTI=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/icons
+    ${if colorVariants != [] then ''
+    name= cp -r \
+      ${lib.optionalString (colorVariants != []) toString colorVariants} \
+      $out/share/icons/
+    ''
+    else ''
+      cp -rf MB-Blueberry-Suru-GLOW MB-Cherry-Suru-GLOW MB-Lime-Suru-GLOW MB-Mango-Suru-GLOW MB-Pistachio-Suru-GLOW MB-Plum-Suru-GLOW Material-Black-Blueberry-Numix-FLAT Material-Black-Blueberry-Numix Material-Black-Blueberry-Suru Material-Black-Cherry-Numix-FLAT Material-Black-Cherry-Numix Material-Black-Cherry-Suru Material-Black-Lime-Numix-FLAT Material-Black-Lime-Numix Material-Black-Lime-Suru Material-Black-Mango-Numix-FLAT Material-Black-Mango-Numix Material-Black-Mango-Suru Material-Black-Pistachio-Numix-FLAT Material-Black-Pistachio-Numix Material-Black-Pistachio-Suru Material-Black-Plum-Numix-FLAT Material-Black-Plum-Numix Material-Black-Plum-Suru $out/share/icons/
```

Comment:
Can we dedupe this list with the one at the top and add some line breaks to short the line?

Reply:
> Can we dedupe this list with the one at the top and add some line breaks to short the line?

Dedupe the list how? Do you have an idea in mind?

Reply:
Not the list itself but the definition near the top of the file.

Reply:
> Not the list itself but the definition near the top of the file.

Something like this?
```nix
  iconset = "
    MB-Blueberry-Suru-GLOW \
    MB-Cherry-Suru-GLOW \
    MB-Lime-Suru-GLOW \
    MB-Mango-Suru-GLOW \
    MB-Pistachio-Suru-GLOW \
    MB-Plum-Suru-GLOW \
    Material-Black-Blueberry-Numix-FLAT \
    Material-Black-Blueberry-Numix \
    Material-Black-Blueberry-Suru \
    Material-Black-Cherry-Numix-FLAT \
    Material-Black-Cherry-Numix \
    Material-Black-Cherry-Suru \
    Material-Black-Lime-Numix-FLAT \
    Material-Black-Lime-Numix \
    Material-Black-Lime-Suru \
    Material-Black-Mango-Numix-FLAT \
    Material-Black-Mango-Numix \
    Material-Black-Mango-Suru \
    Material-Black-Pistachio-Numix-FLAT \
    Material-Black-Pistachio-Numix \
    Material-Black-Pistachio-Suru \
    Material-Black-Plum-Numix-FLAT \
    Material-Black-Plum-Numix \
    Material-Black-Plum-Suru \
  ";
...
    else ''
      cp -rf "${iconset}" $out/share/icons/
    ''
    }

    runHook postInstall
  '';
  ```

Reply:
@SuperSandro2000 what about this solution?
```nix
{ lib
, stdenvNoCC
, fetchFromGitHub
, colorVariants ? []
}:

let
  pname = "material-black-colors";
  colorVariantList = [
    "MB-Blueberry-Suru-GLOW"
    "MB-Cherry-Suru-GLOW"
    "MB-Lime-Suru-GLOW"
    "MB-Mango-Suru-GLOW"
    "MB-Pistachio-Suru-GLOW"
    "MB-Plum-Suru-GLOW"
    "Material-Black-Blueberry-Numix-FLAT"
    "Material-Black-Blueberry-Numix"
    "Material-Black-Blueberry-Suru"
    "Material-Black-Cherry-Numix-FLAT"
    "Material-Black-Cherry-Numix"
    "Material-Black-Cherry-Suru"
    "Material-Black-Lime-Numix-FLAT"
    "Material-Black-Lime-Numix"
    "Material-Black-Lime-Suru"
    "Material-Black-Mango-Numix-FLAT"
    "Material-Black-Mango-Numix"
    "Material-Black-Mango-Suru"
    "Material-Black-Pistachio-Numix-FLAT"
    "Material-Black-Pistachio-Numix"
    "Material-Black-Pistachio-Suru"
    "Material-Black-Plum-Numix-FLAT"
    "Material-Black-Plum-Numix"
    "Material-Black-Plum-Suru"
  ];

in
lib.checkListOfEnum "${pname}: color variants" colorVariantList colorVariants

stdenvNoCC.mkDerivation rec {
  inherit pname;
  version = "unstable-2020-12-17";

  src = fetchFromGitHub {
    owner = "rtlewis88";
    repo = "rtl88-Themes";
    rev = "3864d851aac7f4e76cf23717aee104de234aef74";
    hash = "sha256-BUJMd6Ltq16/HqqDbB5VDGIRSzLivXxNYZPT9sd6oTI=";
  };

  installPhase = ''
    runHook preInstall
    mkdir -p $out/share/icons
    ${if colorVariants != [] then ''
    name= cp -r \
      ${lib.optionalString (colorVariants != []) toString colorVariants} \
      $out/share/icons/
    ''
    else ''
      cp -rf ${lib.concatStringsSep " " colorVariantList} $out/share/icons/
    ''}
    runHook postInstall
  '';

  meta = with lib; {
    description = "Material Black Colors icons";
    homepage = "https://github.com/rtlewis88/rtl88-Themes/tree/material-black-COLORS";
    maintainers = with maintainers; [ d3vil0p3r ];
    platforms = platforms.all;
    license = licenses.gpl3Plus;
  };
}
```

Reply:
yes, like this 👍🏼

Reply:
> yes, like this 👍🏼

Fixed.